service/src/enclave/api.rs

```diff
@@ -103,7 +107,11 @@ pub fn enclave_init(config: &Config) -> EnclaveResult<Enclave> {
 
 	// create an enclave API and initialize it
 	let enclave_api = Enclave::new(enclave);
-	enclave_api.init(&config.mu_ra_url_external(), &config.untrusted_worker_url_external())?;
+	enclave_api.init(
+		&config.mu_ra_url_external(),
+		&config.untrusted_worker_url_external(),
+		&config.base_dir().display().to_string(),
```

Comment:
Isn't `.display()` enough?

Reply:
Nope, you get either:

```bash
error[E0599]: the method `to_string` exists for reference `&std::path::Path`, but its trait bounds were not satisfied
   --> service/src/enclave/api.rs:111:22
    |
111 |         &config.data_dir().to_string(),
    |                            ^^^^^^^^^
    |
    = note: the following trait bounds were not satisfied:
            `std::path::Path: std::fmt::Display`
```

or

```bash
 service/src/enclave/api.rs:111:3
    |
108 |     enclave_api.init(
    |                 ---- arguments to this function are incorrect
...
111 |         &config.data_dir().display(),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `str`, found struct `std::path::Display`
    |
    = note: expected reference `&str`
               found reference `&std::path::Display<'_>`
note: associated function defined here
   --> /opt/shared/integritee-worker/core-primitives/enclave-api/src/enclave_base.rs:36:5
    |
```

Reply:
Read up the docs regarding this, bottom line is you are right, not every `Path` object can be treated as `String`. :face_exhaling: 


>A [Path](https://doc.rust-lang.org/stable/std/path/struct.Path.html) might contain non-Unicode data. This struct implements the [Display](https://doc.rust-lang.org/stable/std/fmt/trait.Display.html) trait in a way that mitigates that.

https://doc.rust-lang.org/stable/std/path/struct.Display.html